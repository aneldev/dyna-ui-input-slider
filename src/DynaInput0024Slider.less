.dyna-input-0024-slider{

  &__label{
    display: flex;
    margin-bottom: 4px;

    &__content{
      flex: auto;
    }
    &__value{
      flex: 0 0;
      white-space: nowrap;
      font-weight: bold;
      font-size: 21px;
      margin-left: 8px;
    }
  }

  // size
  &--size- {
    &PX12 {
      .dyna-slider-daylight__min,
      .dyna-slider-daylight__max {
        display: none;
      }
    }
    &PX24 {
      .dyna-slider-daylight__min,
      .dyna-slider-daylight__max {
        display: none;
      }
    }
    &PX32 {
      .dyna-slider-daylight {
        position: relative;
        top: -1px;
        left: 0;
        font-size: 10px;
        padding: 0 1px;
      }
    }
    &PX48 {
      .dyna-slider-daylight__min,
      .dyna-slider-daylight__max {
        font-size: 14px;
        padding: 0 2px;
      }
    }
  }
}