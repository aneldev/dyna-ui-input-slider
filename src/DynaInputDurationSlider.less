.dyna-input-duration-slider{
  &__label{
    display: flex;
    margin-bottom: 4px;

    &__content{
      flex: auto;
    }
    &__value{
      flex: 0 0;
      white-space: nowrap;
      font-weight: bold;
      font-size: 21px;
      margin-left: 8px;
    }
  }
}