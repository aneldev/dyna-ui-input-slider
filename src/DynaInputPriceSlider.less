.dyna-input-price-slider{
  &__label{
    display: flex;
    margin-bottom: 4px;

    &__content{
      flex: auto;
    }
    &__value{
      flex: 0 0;
      white-space: nowrap;
      font-weight: bold;
      font-size: 21px;
      margin-left: 8px;
    }
  }
}