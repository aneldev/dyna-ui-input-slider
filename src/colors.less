@import (reference) "~dyna-ui-styles/styles/WHITE_BLACK";
@import (reference) "~dyna-ui-styles/styles/WHITE_RED";
@import (reference) "~dyna-ui-styles/styles/RED_WHITE";
@import (reference) "~dyna-ui-styles/styles/WHITE_ORANGE";
@import (reference) "~dyna-ui-styles/styles/BLACK_WHITE";
@import (reference) "~dyna-ui-styles/styles/BLACK_ORANGE";
@import (reference) "~dyna-ui-styles/styles/GREY_WHITE";
@import (reference) "~dyna-ui-styles/styles/ORANGE_WHITE";
@import (reference) "~dyna-ui-styles/colors/pantone-coated";

.dyna-slider {
	&--color- {

		&BLACK_WHITE {
			.color-slider(@color-BLACK_WHITE-foreground-1, @color-BLACK_WHITE-foreground-2)
		}
		&BLACK_ORANGE {
			.color-slider(@color-BLACK_ORANGE-foreground-1, @color-BLACK_ORANGE-foreground-2)
		}
		&TRANSPARENT_ORANGE {
			.color-slider(@color-BLACK_ORANGE-foreground-1, @color-BLACK_ORANGE-foreground-2)
		}
		&TRANSPARENT_WHITE {
			.color-slider(@color-BLACK_WHITE-foreground-1, @color-BLACK_WHITE-foreground-2)
		}
		&ORANGE_WHITE {
			.color-slider(@color-ORANGE_WHITE-foreground-1, @color-ORANGE_WHITE-foreground-2)
		}
		&RED_WHITE {
			.color-slider(@color-RED_WHITE-foreground-1, @color-RED_WHITE-foreground-2)
		}
		&GREY_WHITE {
			.color-slider(@color-GREY_WHITE-foreground-1, @color-GREY_WHITE-foreground-2)
		}
		&WHITE_BLACK {
			.color-slider(@color-WHITE_BLACK-foreground-1, @color-WHITE_BLACK-foreground-2)
		}
		&WHITE_RED {
			.color-slider(@color-WHITE_RED-foreground-1, @color-WHITE_RED-foreground-2)
		}
		&WHITE_ORANGE {
			.color-slider(@color-WHITE_ORANGE-foreground-1, @color-WHITE_ORANGE-foreground-2)
		}
	}
}

.color-slider(@slider-color,@line-color) {
	.rc-slider-handle {
		border-color: @slider-color;
		background-color: @slider-color;

		&:active {
			border-color: @slider-color;
			box-shadow: none;
		}

		&:hover {
			border-color: @slider-color;
		}
	}

	.rc-slider-track {
		background-color: @line-color;
	}

}

