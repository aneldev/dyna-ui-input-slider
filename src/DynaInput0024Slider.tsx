import * as React from "react";
import {dynaClassName, DynaClassName} from "dyna-class-name"
import {EColor} from "dyna-ui-styles";

import {DynaInputRangeSlider} from "./DynaInputRangeSlider";
import {ESize, IHourRange} from "./interfaces";

import {StatsBar} from "./components/StatsBar";
import {Daylight} from "./components/Daylight";

import "./DynaInput0024Slider.less";

export interface IDynaInput0024SliderProps {
  className?: string;
  name?: string;
  color?: EColor;
  size?: ESize;
  label?: JSX.Element;
  value: IHourRange;        // hour: 00-24
  statsHours?: number [];   // for stats, numbers 0-24
  onChange?: (name: string, value: IHourRange) => void;
}

export class DynaInput0024Slider extends React.Component<IDynaInput0024SliderProps> {
  static defaultProps: IDynaInput0024SliderProps = {
    className: "",
    name: null,
    label: null,
    color: EColor.WHITE_BLACK,
    size: ESize.PX24,
    value: {from: 0, to: 24},
    onChange: (name: String, value: IHourRange) => undefined,
  };

  private readonly className: DynaClassName = dynaClassName("dyna-input-0024-slider");

  private handleChange(name: string, values: number[]): void {
    const {onChange} = this.props;
    onChange(name, {from: values[0], to: values[1]})
  }

  private renderLabel(): JSX.Element {
    const {
      label,
      value: {
        from, to,
      },
    } = this.props;

    return (
      <div className={this.className("__label")}>
        <div className={this.className("__label__content")}>{label}</div>
        <div className={this.className("__label__value")}>{`${from}:00 - ${to}:00`}</div>
      </div>
    );
  }

  private renderTopBackground(): JSX.Element {
    const {statsHours} = this.props;
    return (
      <StatsBar
        stats={statsHours}
      />
    );
  }

  private renderBottomBackground(): JSX.Element{
    return (
      <Daylight />
    );
  }

  public render(): JSX.Element {
    const {
      className: userClassName,
      name,
      color,
      size,
      value: {
        from, to,
      },
    } = this.props;

    const className: string = this.className("", userClassName && '/' + userClassName || "")

    return (
      <div className={className}>
        {this.renderLabel()}
        <DynaInputRangeSlider
          name={name}
          color={color}
          size={size}
          topBackground={this.renderTopBackground()}
          bottomBackground={this.renderBottomBackground()}
          min={0}
          max={24}
          pushable
          value={[from, to]}
          onChange={this.handleChange.bind(this)}/>
      </div>
    );
  }
}
