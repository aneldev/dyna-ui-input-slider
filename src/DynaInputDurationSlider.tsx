import * as React from "react";
import {DynaClassName, dynaClassName} from "dyna-class-name";

import {EColor} from "dyna-ui-styles";
import {EMin, ESize} from "./interfaces";

import {DynaInputSlider} from "./DynaInputSlider";
import {StatsBar} from "./components/StatsBar";
import {StatsHelper} from "./utils/StatsHelper";

import "./DynaInputDurationSlider.less";

export interface IDynaInputDurationSliderProps {
  className?: string;
  name?: string;
  label?: JSX.Element | string;
  color?: EColor;
  size?: ESize;
  ticksCount?: number;
  values: number[];       // for stats and to get the min/max
  unitSuffix?: string;          // unit suffix, used for ui only
  minType: EMin;
  value: number;
  onChange: (name: string, value: number) => void;
}

export class DynaInputDurationSlider extends React.Component<IDynaInputDurationSliderProps> {
  static defaultProps: IDynaInputDurationSliderProps = {
    className: undefined,
    name: null,
    unitSuffix: 'h',
    label: null,
    color: EColor.WHITE_BLACK,
    size: ESize.PX24,
    ticksCount: 24,
    values: [0, 200],
    minType: EMin.ZERO,
    value: 0,
    onChange: (name: string, value: number) => undefined,
  };

  constructor(props: IDynaInputDurationSliderProps) {
    super(props);
    this.statsHelper.setData(props.values);
  }

  public componentWillReceiveProps(nextProps:IDynaInputDurationSliderProps):void{
    this.statsHelper.setData(nextProps.values);
  }

  private readonly className = dynaClassName("dyna-input-duration-slider");
  private readonly statsHelper: StatsHelper = new StatsHelper();

  private handleChange(name: string, value: number): void {
    const {onChange} = this.props;
    onChange(name, value)
  }

  private getStatTicks(): number[] {
    const {minType, ticksCount} = this.props;
    return this.statsHelper.getFloatGroupTicks(minType, ticksCount);
  }

  private renderTopBackground(): JSX.Element {
    if (!this.statsHelper.hasValues) return null;
    return <StatsBar ticks={this.getStatTicks()}/>;
  }

  private renderBottomBackground(): JSX.Element {
    const {minType, unitSuffix} = this.props;
    const csMinMax: DynaClassName = dynaClassName(this.className("__min-max-container"));
    return (
      <div className={csMinMax("")}>
        <div className={csMinMax("__min")}>{`${this.statsHelper.getMinValue(minType)}${unitSuffix}`}</div>
        <div className={csMinMax("__max")}>{`${this.statsHelper.getMaxValue()}${unitSuffix}`}</div>
      </div>
    );
  }

  private renderLabel(): JSX.Element {
    const {
      label,
      unitSuffix,
      value,
    } = this.props;

    return (
      <div className={this.className("__label")}>
        <div className={this.className("__label__content  /dyna-slider-label")}>{label}</div>
        <div className={this.className("__label__value /dyna-slider-value")}>{`${value}${unitSuffix}`}</div>
      </div>
    );
  }

  public render(): JSX.Element {
    const {
      className: userClassName,
      name, color, size,
      minType, value,
    } = this.props;

    const className: string = `--size-${size}`;

    return (
      <div className={this.className.root(this.props, className)}>
        {this.renderLabel()}
        <DynaInputSlider
          name={name}
          color={color}
          size={size}
          min={this.statsHelper.getMinValue(minType)}
          max={this.statsHelper.getMaxValue()}
          topBackground={this.renderTopBackground()}
          bottomBackground={this.renderBottomBackground()}
          value={value}
          onChange={this.handleChange.bind(this)}
        />
      </div>
    );
  }
}
