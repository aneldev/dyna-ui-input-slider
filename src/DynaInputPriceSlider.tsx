import * as React from "react";
import {DynaClassName, dynaClassName} from "dyna-class-name";
import {EColor} from "dyna-ui-styles";
import {round} from "dyna-loops";

import {EMin, ESize} from "./interfaces";

import {DynaInputSlider} from "./DynaInputSlider";
import {StatsBar} from "./components/StatsBar";

import {getMaxValue, getMinValue, getTicks} from "./utils";

import "./DynaInputPriceSlider.less";

export interface IDynaInputPriceSliderProps {
  className?: string
  name?: string;
  color?: EColor;
  size?: ESize;
  label?: JSX.Element;
  prices: number[];
  step?: number;
  statTicksCount?: number;
  minType: EMin;
  value: number;
  formatPrice?: (value: number) => string;
  onChange: (name: string, value: number) => void;
}

export class DynaInputPriceSlider extends React.Component<IDynaInputPriceSliderProps> {
  static defaultProps: IDynaInputPriceSliderProps = {
    className: "",
    name: null,
    label: null,
    color: EColor.WHITE_BLACK,
    size: ESize.PX24,
    prices: [],
    step: 10,
    statTicksCount: 24,
    minType: EMin.ZERO,
    value: 0,
    formatPrice: (value: number) => `$${value}`,
    onChange: (name: string, value: number) => undefined,
  };

  private readonly className = dynaClassName("dyna-input-price-slider");

  private get minPrice(): number {
    const {minType, prices} = this.props;
    return Math.floor(getMinValue(minType, prices))
  }

  private get maxPrice(): number {
    const {step, prices} = this.props;
    return Math.ceil(getMaxValue(prices) + step);
  }

  private handleChange(name: string, value: number): void {
    const {onChange} = this.props;
    onChange(name, value)
  }

  private renderTopBackground(): JSX.Element {
    const {prices, statTicksCount} = this.props;
    return <StatsBar ticks={getTicks(prices, statTicksCount)}/>;
  }

  private renderBottomBackground(): JSX.Element {
    const {formatPrice} = this.props;
    const csMinMax: DynaClassName = dynaClassName(this.className("__min-max-container"));
    return (
      <div className={csMinMax("")}>
        <div className={csMinMax("__min")}>{formatPrice(this.minPrice)}</div>
        <div className={csMinMax("__max")}>{formatPrice(this.maxPrice)}</div>
      </div>
    );
  }

  private renderLabel(): JSX.Element {
    const {
      label,
      formatPrice,
      value,
    } = this.props;

    return (
      <div className={this.className("__label")}>
        <div className={this.className("__label__content")}>{label}</div>
        <div className={this.className("__label__value")}>{formatPrice(round(value, -1))}</div>
      </div>
    );
  }

  public render(): JSX.Element {
    const {
      className: userClassName,
      name,
      color, size,
      step,
      value,
    } = this.props;

    const className: string = this.className(
      "",
      userClassName && "/" + userClassName,
      `--size-${size}`,
    );

    return (
      <div className={className}>
        {this.renderLabel()}
        <DynaInputSlider
          name={name}
          color={color}
          size={size}
          step={step}
          min={this.minPrice}
          max={this.maxPrice}
          topBackground={this.renderTopBackground()}
          bottomBackground={this.renderBottomBackground()}
          value={round(value, -1)}
          onChange={this.handleChange.bind(this)}
        />
      </div>
    );
  }
}
