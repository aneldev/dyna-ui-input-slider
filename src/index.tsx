import 'rc-slider/assets/index.css';
import "./layout.less";
import "./colors.less";

export {
  DynaSlider, IDynaSliderProps,
} from "./DynaSlider";

export {
  DynaRangeSlider, IDynaRangeSliderProps,
} from "./DynaRangeSlider";


export {
  EColor,
} from "dyna-ui-styles";
