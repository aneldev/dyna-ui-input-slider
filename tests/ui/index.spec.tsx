import "jest";

import { configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import * as React from 'react';
import * as enzyme from 'enzyme';

import {DynaInputSlider} from '../../src';

describe('Home', () => {
  let wrapper;

  it('has expected content with deep render', () => {
    wrapper = enzyme.shallow(
      (
        <DynaInputSlider
          name="power"
          min={0}
          max={100}
          value={50}
          onChange={() => undefined}
        />
      ),
      {}
    );

    expect(wrapper).toMatchSnapshot();
  });
});
