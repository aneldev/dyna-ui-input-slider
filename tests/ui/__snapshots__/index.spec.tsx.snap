// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home has expected content with deep render 1`] = `
<div
  className="dyna-slider  dyna-slider--single-value-mode dyna-slider--color-WHITE_BLACK dyna-slider--size-PX24"
>
  <ComponentEnhancer(undefined)
    activeDotStyle={Object {}}
    className=""
    disabled={false}
    dotStyle={Object {}}
    dots={false}
    handle={[Function]}
    handleStyle={
      Array [
        Object {},
      ]
    }
    included={true}
    marks={Object {}}
    max={100}
    min={0}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    railStyle={Object {}}
    step={1}
    trackStyle={
      Array [
        Object {},
      ]
    }
    value={50}
    vertical={false}
  />
</div>
`;
